File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    void reverseString(vector<char>& vec) {
        auto its = vec.begin();
        auto ite = vec.end()-1;
        while(its < ite){
            char temp = *its;
            *its = *ite;
            *ite = temp;
            its++;
            ite--;
        }    
    }
};

File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        return self.reverse(s, 1)

    def reverse(self, s : List[str], num : int) -> None:
        if(num>len(s)):
             return
        ch = s[num-1]
        self.reverse(s,num+1)
        s[len(s)-num] = ch